src/dwarf/function.rs

```diff
@@ -112,6 +114,130 @@ where
 }
 
 
+pub(super) struct InlinedFunction<'dwarf> {
+    pub(crate) dw_die_offset: gimli::UnitOffset<<R<'dwarf> as gimli::Reader>::Offset>,
+    pub(crate) name: Option<R<'dwarf>>,
+    pub(crate) call_file: Option<u64>,
+    pub(crate) call_line: u32,
+    pub(crate) call_column: u32,
+}
+
+
+struct InlinedFunctionAddress {
+    range: gimli::Range,
+    call_depth: usize,
+    /// An index into `Function::inlined_functions`.
+    function: usize,
+}
+
+
+pub(super) struct InlinedFunctions<'dwarf> {
+    /// List of all `DW_TAG_inlined_subroutine` details in this
+    /// function.
+    inlined_functions: Box<[InlinedFunction<'dwarf>]>,
+    /// List of `DW_TAG_inlined_subroutine` address ranges in this
+    /// function.
+    inlined_addresses: Box<[InlinedFunctionAddress]>,
+}
+
+impl<'dwarf> InlinedFunctions<'dwarf> {
+    pub(crate) fn parse(
+        dw_die_offset: gimli::UnitOffset<<R<'dwarf> as gimli::Reader>::Offset>,
+        unit: &gimli::Unit<R<'dwarf>>,
+        sections: &gimli::Dwarf<R<'dwarf>>,
+    ) -> Result<Self, Error> {
+        let mut entries = unit.entries_raw(Some(dw_die_offset))?;
+        let depth = entries.next_depth();
+        // The DIE offset we get is the one of the function. But we are
+        // interested in its children. So skip the necessary attributes.
+        let abbrev = entries.read_abbreviation().unwrap().unwrap();
```

Comment:
is `.unwrap().unwrap()` safe to do here?

Reply:
It should be. At this point we have parsed the function already and we just redo these steps. Nevertheless, the read itself could probably short circuit more gracefully. Will adjust.